src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2941,6 +2941,18 @@ private bool isExactType(CORINFO_CLASS_STRUCT_* cls)
             return _compilation.IsEffectivelySealed(type);
         }
 
+        private TypeCompareState isGenericType(CORINFO_CLASS_STRUCT_* cls)
+        {
+            TypeDesc type = HandleToObject(cls);
+
+            if (type.IsCanonicalSubtype(CanonicalFormKind.Any))
```

Comment:
```suggestion
            if (type is CanonBaseType)
```
IsCanonicalSubtype is too conservative. @MichalStrehovsky Is there a better way to say `type is CanonBaseType` ?

Reply:
```suggestion
            if (type.IsCanonicalDefinitionType(CanonicalFormKind.Any))
```